Source/WebGPU/WebGPU/XRSubImage.mm

```diff
@@ -36,12 +36,12 @@
 namespace WebGPU {
 
 XRSubImage::XRSubImage(bool, Device& device)
-    : m_device(device)
+    : m_device(&device)
 {
 }
 
 XRSubImage::XRSubImage(Device& device)
-    : m_device(device)
+    : m_device(&device)
 {
 }
```

Comment:
I think `ThreadSafeWeakPtr` has a constructor that works with a const reference to an object that inherits from `ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr`, so you might not need the `&`s.